### PR TITLE
add Aleph One Marathon Games

### PR DESCRIPTION
**Pull request by theofficialgman**
closes https://github.com/Botspot/pi-apps/issues/1450

@Rak1ta thank you for the very well written scripts. they needed some minor editing but they mostly just worked.

I removed boost armhf hardcoded boost location. can you explain why you added that? you can try running this PR by importing it to pi-apps (using the pi-apps settings import option) and send a logfile here if it fails.

### CODE REVIEW

**Comment by theofficialgman** [comment]
also before @Botspot asks, no the uninstall doesn't delete savedata. that is stored here: `~/.alephone`

**Comment by Rak1ta** [comment]
> boost armhf

At the time when I installed the game, I had an error during installation.

**Review by Rak1ta — CHANGES_REQUESTED** [review]
```
line 14_15:rm -rf ~/AlephOne-${version} ~/AlephOneData ~/AlephOne-${version}.tar.bz2 ~/rm -rf Marathon*-${version}-Data.zip
line 64: rm -rf Marathon*-${version}-Data.zip
```
For pi-apps to delete only what it uploaded?

**Comment by theofficialgman** [comment]
> ```
> line 14_15:rm -rf ~/AlephOne-${version} ~/AlephOneData ~/AlephOne-${version}.tar.bz2 ~/rm -rf Marathon*-${version}-Data.zip
> line 64: rm -rf Marathon*-${version}-Data.zip
> ```
> 
> For pi-apps to delete only what it uploaded?

I don't get what you are saying here... can you make the proper suggestion? you seem to have messed up the formatting

**Comment by theofficialgman** [comment]
> At the time when I installed the game, I had an error during installation. And I didn't change this item again.

ah I see the problem... the configure script is falsely detecting the system as arm64 when using arm64 kernel on armhf OS
```
checking build system type... aarch64-unknown-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking target system type... aarch64-unknown-linux-gnu
```

**Comment by theofficialgman** [comment]
ok should work now on armhf systems @Rak1ta its work a try again

**Comment by Rak1ta** [comment]
@theofficialgman so that you do not wait, I will be able to check in 8 hours, that is, late. Please https://github.com/Botspot/pi-apps/pull/1874#pullrequestreview-984156045[check](https://github.com/Botspot/pi-apps/pull/1874#pullrequestreview-984156045), either only downloaded files are deleted or all previously downloaded game files must be deleted.

**Comment by theofficialgman** [comment]
I don't really understand what you are saying there. the script is correct. it deletes any files that pi-apps installs without touching the save data (that is in a separate folder)

**Comment by theofficialgman** [comment]
oh are you just saying that this is unecessary? `rm -f ~/AlephOne-${version}.tar.bz2 ~/Marathon-${version}-Data.zip ~/Marathon2-${version}-Data.zip ~/MarathonInfinity-${version}-Data.zip` since I moved the download to take place in /tmp

**Comment by Rak1ta** [comment]
@theofficialgman 👍
line 113: $HOME/.wine - I made a typo, forgot to change. 
If no one checks before, I'll try on a new Buster installation

File: apps/Marathon/uninstall [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+#game files and shortcuts
+rm -f $HOME/.local/share/applications/marathon*.desktop
+rm -rf $HOME/AlephOneData
+
+#executable file
+sudo rm -rf /usr/local/bin/alephone /usr/local/share/AlephOne
+
+echo -e "\n\n\e[93mYou just uninstalled the Aleph One app, but it's not completely gone yet.
+Setup files is located in the $HOME/ folder.
```
Comment (Crilum) [inline apps/Marathon/uninstall:11]:
```suggestion
Setup files are located in the $HOME/ folder.
```

Comment (Rak1ta) [reply apps/Marathon/uninstall:11]: $HOME/AlephOneData

**Comment by Rak1ta** [comment]
@theofficialgman successful launch Debian Buster 32
[install-success-Marathon.log](https://github.com/Botspot/pi-apps/files/8771214/install-success-Marathon.log)
[uninstall-success-Marathon.log](https://github.com/Botspot/pi-apps/files/8771216/uninstall-success-Marathon.log)

**Comment by theofficialgman** [comment]
> @theofficialgman successful launch Debian Buster 32, but with modification: `Exec=/usr/local/bin/arm-linux-gnueabihf-alephone` - need this file path for icons.

icons work, they come directly from the pi-apps marathon icon.

>  but with modification: `Exec=/usr/local/bin/arm-linux-gnueabihf-alephone`

this is not for icons at all... this is the executable location of the alephone binary... 
```
/usr/bin/install -c alephone '/usr/local/bin/./arm-linux-gnueabihf-alephone'
```
no clue why its installing there... I'll have to figure that out.

> ffmpeg [dependency](https://github.com/Aleph-One-Marathon/alephone/commit/580adb8037dd9ea385bcdfe283ff804d4091ffe9) required? I have it installed without it.

ffmpeg is an optional dependency:
```
If FFmpeg support is enabled, it will be used instead of libmad, libsndfile, libvorbis, and SMPEG. Any errors with those libraries can be ignored. If FFmpeg is disabled or unavailable, those other libraries will be used to support various media types and features. 
```

**Comment by theofficialgman** [comment]
ok I think that is solved. rebuilding now on my armhf chroot to test

**Comment by theofficialgman** [comment]
alright I build and tested, it all works now. opening up for merging

**Comment by theofficialgman** [comment]
ok added a custom updater (since this has 4 files but only one version). all seems good

File: apps/Marathon/install [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+cd AlephOne-${version}
+
+# get the $DISTRIB_RELEASE by calling lsb_release
+# check if upstream-release is available
+if [ -f /etc/upstream-release/lsb-release ]; then
+  echo "This is a Ubuntu Derivative, checking the upstream-release version info"
+  DISTRIB_CODENAME=$(lsb_release -s -u -c)
+else
+  DISTRIB_CODENAME=$(lsb_release -s -c)
+fi
```
Comment (Botspot) [inline apps/Marathon/install:38]: Should these lines be replaced with the `get_codename` function?

Comment (theofficialgman) [reply apps/Marathon/install:38]: not sure I like the get_codename function... I worried that the apt output would pollute the variable with stdout
```
get_codename() { #get debian/ubuntu codename
  if ! command -v lsb_release &>/dev/null; then
    apt_update && sudo apt-get install -y lsb-release
  fi
  
  # check if upstream-release is available
  if [ -f /etc/upstream-release/lsb-release ]; then
    # this is a derivative, get the upstream release info
    lsb_release -suc
else
    lsb_release -sc
  fi
}
```

Comment (theofficialgman) [reply apps/Marathon/install]: I verified get_codename is a fine replacement, switching now

**Comment by theofficialgman** [comment]
@Botspot do you have any concerns with these games? should be ready for merge

**Comment by Botspot** [comment]
Just tested the original "Marathon" game, and found it mostly unplayable.
The slightest mouse movement rotated the in-game camera by a lot, so I found it nearly impossible to face the right way quickly.

https://user-images.githubusercontent.com/54716352/174347513-66379fda-7695-41ad-9cee-412f2354703c.mp4

Also, the frame rate stayed around 6fps when not screen recording.

**Comment by Rak1ta** [comment]
@Botspot What Pi do you have? It works great on mine, but I set the resolution to 960x720, I don't like full screen in such games.

**Comment by theofficialgman** [comment]
> Just tested the original "Marathon" game, and found it mostly unplayable. The slightest mouse movement rotated the in-game camera by a lot, so I found it nearly impossible to face the right way quickly.

this is not a bug... I guess I have to explain or link you to "raw mouse input"
the basis of it is that for games, obtaining the raw mouse input is MUCH better that grabbing mouse data from xorg or other sources which introduce their own smoothing and acceleration curves. your raw mouse input movements will be very large if you have a mouse with a large DPI. most bargin bin mice have low dpi and will have low movement, while nice mice have high dpi and will have high movement. you will have to go to the in game settings to adjust the sensitivity.

**Comment by theofficialgman** [comment]
regarding the performance, I don't know how well it runs on a pi (Rak1ta indicates its good)

probably best to check that it isn't falling back to software rendering or something by running from a terminal
```
/usr/local/bin/alephone /home/garrett/AlephOneData/Marathon/
```
and of course checking preferences -> graphics to make sure you aren't trying to play at 4K or something

**Comment by Botspot** [comment]
Hmm...
```
libGL error: failed to create dri screen
libGL error: failed to load driver: vc4
libGL error: failed to create dri screen
libGL error: failed to load driver: vc4
while initializing preferences (preferences.cpp:3212)
  Reading older preferences of version 20200803. Preferences will be upgraded to version 20220115 when saved. (/home/pi/AlephOneData/Marathon/Scripts/Default Preferences.xml) (preferences.cpp:3306)
GL_VENDOR: Mesa/X.org
GL_RENDERER: llvmpipe (LLVM 11.0.1, 128 bits)
GL_VERSION: 3.1 Mesa 20.3.5
```
llvmpipe means software rendering. I'm not sure what could be causing this.

**Comment by Botspot** [comment]
full kms is enabled btw.

**Comment by theofficialgman** [comment]
> full kms is enabled btw.

I was under the impression that under some raspbian/debian release, hardware accelerated opengl did not work at all under one of the KMS options.

what is full kms? I don't think that exists. unless you are using it to describe KMS, and not FKMS (Fake/Firmware KMS)

not sure which release combination that is, but I have read it before that KMS has no hardware accelerated graphics at all on the pi4.... why can't the pi devs just make something that works everywhere

**Comment by theofficialgman** [comment]
ah the sticky explains it well https://forums.raspberrypi.com/viewtopic.php?t=317511

how complicated

**Comment by Botspot** [comment]
I must have messed something up. Just launched DDNet again and now it's really slow too. On my system a few days ago it was fast.

Will reboot and see if that improves things.

**Comment by Botspot** [comment]
> Will reboot and see if that improves things.

It did. Game is now quite usable after turning down mouse settings.

**Pull request merged**